src/components/textInput/TextInputBase.tsx

```diff
@@ -328,6 +333,20 @@ export const TextInputBase = ({
       const actualTextLength =
         inputType !== "default" ? text.replace(/\s/g, "").length : text.length;
 
+      // Notify the user when the limit is reached
+      // This is only for iOS, as Android handles it via accessibilityLiveRegion
+      if (
+        counterLimit &&
+        actualTextLength >= counterLimit &&
+        accessibilityAnnounceLimitReached
+      ) {
+        if (Platform.OS === "ios") {
+          // Su iOS, utilizzare accessibilityAnnouncement per notificare
+          AccessibilityInfo.announceForAccessibility(
+            accessibilityAnnounceLimitReached
+          );
+        }
+      }
```

Comment:
just a micro suggestion: if all the conditions are AND, we could join them in a single statement. Also the comment is redundant because you specified everything in the line above
```suggestion
      if (
        counterLimit &&
        actualTextLength >= counterLimit &&
        accessibilityAnnounceLimitReached &&
        Platform.OS === "ios"
      ) {
          AccessibilityInfo.announceForAccessibility(
            accessibilityAnnounceLimitReached
          );
      }
```